Lib/test/libregrtest/main.py

```diff
@@ -708,7 +711,12 @@ def _init(self):
 
         strip_py_suffix(self.cmdline_args)
 
-        self.tmp_dir = get_temp_dir(self.tmp_dir)
+        self._tmp_dir = get_temp_dir(self._tmp_dir)
+
+    @property
+    def tmp_dir(self) -> StrPath:
+        assert self._tmp_dir is not None, "self._tmp_dir is None"
+        return self._tmp_dir
```

Comment:
the need for things like this IMO reveals deeper design flaws that we should tackle at some point. But that doesn't have to be done now; enabling `--strict-optional` will make mypy a lot more useful in general, so I agree that it's worth doing in its own right :-)

Also, similar to the above, I'd try to make this assertion message say something about what invariant has been broken if `self._tmp_dir` ends up being `None` unexpectedly